### PR TITLE
llava-next Fp8

### PR DESCRIPTION
**Pull request by yuanwu2017**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by yuanwu2017** [comment]
@mandy-li  @tthakkal 

1. Llava-v1.6-mistral-7b BF16 inference with FusedSDPA
command:
```
MODEL_ID=llava-hf/llava-v1.6-mistral-7b-hf
docker run -it -p 8083:80 \
   --name yuanwu-1.17 \
   --runtime=habana \
   -v $volume:/data \
   -v ~/workspace/hqt_output:/usr/src/hqt_output \
   -v ~/workspace:/workspace \
   -v ~/workspace/quant_config:/configs \
   -e HABANA_VISIBLE_DEVICES=all \
   -e PT_HPU_ENABLE_LAZY_COLLECTIVES=true \
   -e OMPI_MCA_btl_vader_single_copy_mechanism=none \
   -e ENABLE_HPU_GRAPH=true \
   -e LIMIT_HPU_GRAPH=true \
   -e USE_FLASH_ATTENTION=true \
   -e FLASH_ATTENTION_RECOMPUTE=true \
   -e http_proxy=${http_proxy}     -e https_proxy=${https_proxy} -e no_proxy=${no_proxy} \
   --cap-add=sys_nice \
   --ipc=host \
   tgi-yuanwu:fp8 \
   --model-id $MODEL_ID --max-input-tokens 4096 --max-total-tokens 8192 --max-batch-prefill-tokens 8192
```
result:
image
image

2. Llava-v1.6-mistral-7b FP8 inference with FusedSDPA
command:
```
MODEL_ID=llava-hf/llava-v1.6-mistral-7b-hf
docker run -it -p 8083:80 \
   --name yuanwu-1.17 \
   --runtime=habana \
   -v $volume:/data \
   -v ~/workspace/hqt_output:/usr/src/hqt_output \
   -v ~/workspace:/workspace \
   -v ~/workspace/quant_config:/configs \
   -e HABANA_VISIBLE_DEVICES=all \
   -e PT_HPU_ENABLE_LAZY_COLLECTIVES=true \
   -e OMPI_MCA_btl_vader_single_copy_mechanism=none \
   -e QUANT_CONFIG=/configs/maxabs_quant.json \
   -e ENABLE_HPU_GRAPH=true \
   -e LIMIT_HPU_GRAPH=true \
   -e USE_FLASH_ATTENTION=true \
   -e FLASH_ATTENTION_RECOMPUTE=true \
   -e http_proxy=${http_proxy}     -e https_proxy=${https_proxy} -e no_proxy=${no_proxy} \
   --cap-add=sys_nice \
   --ipc=host \
   tgi-yuanwu:fp8 \
   --model-id $MODEL_ID --max-input-tokens 4096 --max-total-tokens 8192 --max-batch-prefill-tokens 8192
```
result:
image
image

File: README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ limitations under the License.
     - [LLama 70b BF16 on 8 Gaudi2 card](#llama-70b-bf16-on-8-gaudi2-card)
     - [LLama 70b FP8 on 8 Gaudi2 card](#llama-70b-fp8-on-8-gaudi2-card)
     - [LLava-next 7B BF16 on 1 Gaudi2 card](#llava-next-7b-bf16-on-1-gaudi2-card)
+    - [LLava-next 7B FP8 on 1 Gaudi2 card](#llava-next-7b-fp8-on-1-gaudi2-card)
```
Comment (mandy-li) [inline README.md]: @yuanwu2017 , there are quite a few variants for Llava-next. Please list those which were supported here, such as this README: https://github.com/huggingface/optimum-habana/tree/main/examples/image-to-text

File: README.md [hunk 2]
```diff
@@ -266,7 +267,43 @@ docker run -p 8080:80 \
...
+   ghcr.io/huggingface/tgi-gaudi:2.0.1 \
+   --model-id $model \
+   --max-input-tokens 4096 \
+   --max-batch-prefill-tokens 16384 \
+   --max-total-tokens 8192
+```
+
+Please implement the inference client according to the [tutorial](https://github.com/huggingface/tgi-gaudi/blob/habana-main/docs/source/basic_tutorials/visual_language_models.md).
+
+### LLava-next 7B FP8 on 1 Gaudi2 card
```
Comment (mandy-li) [inline README.md]: Same comment as README. you may list the name as llava-v1.6-mistral-7b in this case

**Review by mandy-li — CHANGES_REQUESTED** [review]
see my comments

File: README.md [hunk 3]
```diff
@@ -242,6 +250,69 @@ docker run -p 8080:80 \
...
+Copy the quantization_config directory from [optimum-habana](https://github.com/huggingface/optimum-habana/tree/main/examples/image-to-text/quantization_config) into $PWD.
+Generate the hqt_output by running the following example in [optimum-habana example](https://github.com/huggingface/optimum-habana/tree/main/examples/image-to-text). Copy the hqt_output into $PWD.
+```bash
+QUANT_CONFIG=./quantization_config/maxabs_measure.json python run_pipeline.py \
```
Comment (tthakkal) [inline README.md]: we dont mention OH measure command for other models, @mandy-li should we mention for this and all other models too ? or do not mention for this model too ?

Comment (mandy-li) [reply README.md]: No, pls don't mention OH command here. Just say use OH to generate the measurment files as before

Comment (yuanwu2017) [reply README.md]: Remove these lines. I see that they have been mentioned at https://github.com/huggingface/tgi-gaudi?tab=readme-ov-file#running-tgi-with-fp8-precision

**Comment by yuanwu2017** [comment]
Test the llava-next with multi queries at same time.
command:
https://github.com/yuanwu2017/llm-dbg/blob/main/run_tgi_llava_next_benchmark.sh

image
image

File: README.md [hunk 4]
```diff
@@ -242,6 +250,60 @@ docker run -p 8080:80 \
...
+   --cap-add=sys_nice \
+   --ipc=host \
+   ghcr.io/huggingface/tgi-gaudi:2.0.1 \
+   --model-id $model \
+   --max-input-tokens 4096 \
+   --max-batch-prefill-tokens 16384 \
+   --max-total-tokens 8192
+```
+
+Please implement the inference client according to the [tutorial](https://github.com/huggingface/tgi-gaudi/blob/habana-main/docs/source/basic_tutorials/visual_language_models.md).
```
Comment (tthakkal) [inline README.md]: I also tried using curl, and it works for me, can we keep it simple 
`
curl -N 127.0.0.1:8080/generate_stream    -X POST      -d '{"inputs":"What is this a picture of?\n\n","parameters":{"seed": 42}}'     -H 'Content-Type: application/json'
`

Comment (yuanwu2017) [reply README.md]: Done

**Comment by tthakkal** [comment]
@yuanwu2017 I tested `llava-v1.6-mistral-7b-hf` model with 2/8 cards bf16/fp8 and it crashes, below command and error. same command works fine if I use 1 card

```bash

MODEL_ID=llava-hf/llava-v1.6-mistral-7b-hf
docker run -it -p 8083:80 \
   --name yuanwu-1.17 \
   --runtime=habana \
   -v $volume:/data \
   -v ~/workspace:/workspace \
   -e HABANA_VISIBLE_DEVICES=all \
   -e PT_HPU_ENABLE_LAZY_COLLECTIVES=true \
   -e OMPI_MCA_btl_vader_single_copy_mechanism=none \
   -e ENABLE_HPU_GRAPH=true \
   -e LIMIT_HPU_GRAPH=true \
   -e USE_FLASH_ATTENTION=true \
   -e FLASH_ATTENTION_RECOMPUTE=true \
   -e http_proxy=${http_proxy}     -e https_proxy=${https_proxy} -e no_proxy=${no_proxy} \
   --cap-add=sys_nice \
   --ipc=host \
   tgi-yuanwu:fp8 \
   --model-id $MODEL_ID --max-input-tokens 4096 --max-total-tokens 8192 --max-batch-prefill-tokens 8192 \
    --sharded true \
   --num-shard 2

```

```text
2024-08-23T17:33:46.185367Z  INFO text_generation_router: router/src/main.rs:345: Warming up model
2024-08-23T17:33:47.368491Z DEBUG text_generation_launcher: Method Warmup encountered an error.
2024-08-23T17:33:47.368514Z DEBUG text_generation_launcher: Traceback (most recent call last):
2024-08-23T17:33:47.368516Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 1106, in warmup
2024-08-23T17:33:47.368519Z DEBUG text_generation_launcher:     _, prefill_batch, _ = self.generate_token([batches[0]], is_warmup)
2024-08-23T17:33:47.368521Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/contextlib.py", line 79, in inner
2024-08-23T17:33:47.368523Z DEBUG text_generation_launcher:     return func(*args, **kwds)
2024-08-23T17:33:47.368526Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 897, in generate_token
2024-08-23T17:33:47.368528Z DEBUG text_generation_launcher:     batch.logits, batch.past = self.forward(
2024-08-23T17:33:47.368530Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 783, in forward
2024-08-23T17:33:47.368532Z DEBUG text_generation_launcher:     model_inputs = self.model.prepare_inputs_for_generation(**kwargs)
2024-08-23T17:33:47.368534Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/custom_modeling/llava_next.py", line 171, in prepare_inputs_for_generation
2024-08-23T17:33:47.368536Z DEBUG text_generation_launcher:     image_features = self.vision_tower(
2024-08-23T17:33:47.368546Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368548Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368550Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368552Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368554Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 339, in forward
2024-08-23T17:33:47.368557Z DEBUG text_generation_launcher:     return self.vision_model(
2024-08-23T17:33:47.368559Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368561Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368563Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368565Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368567Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 300, in forward
2024-08-23T17:33:47.368569Z DEBUG text_generation_launcher:     encoder_outputs = self.encoder(
2024-08-23T17:33:47.368571Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368573Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368575Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368577Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368579Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 251, in forward
2024-08-23T17:33:47.368582Z DEBUG text_generation_launcher:     layer_outputs = encoder_layer(
2024-08-23T17:33:47.368584Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368586Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368588Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368590Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368591Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 190, in forward
2024-08-23T17:33:47.368594Z DEBUG text_generation_launcher:     hidden_states, attn_weights = self.self_attn(
2024-08-23T17:33:47.368595Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368597Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368599Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368601Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368603Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 158, in forward
2024-08-23T17:33:47.368605Z DEBUG text_generation_launcher:     attn_output = attn_output.reshape(bsz, tgt_len, embed_dim)
2024-08-23T17:33:47.368607Z DEBUG text_generation_launcher: RuntimeError: shape '[3, 577, 1024]' is invalid for input of size 886272
2024-08-23T17:33:47.368609Z DEBUG text_generation_launcher: 
2024-08-23T17:33:47.368610Z DEBUG text_generation_launcher: During handling of the above exception, another exception occurred:
2024-08-23T17:33:47.368612Z DEBUG text_generation_launcher: 
2024-08-23T17:33:47.368614Z DEBUG text_generation_launcher: Traceback (most recent call last):
2024-08-23T17:33:47.368616Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/tgi_service.py", line 37, in <module>
2024-08-23T17:33:47.368618Z DEBUG text_generation_launcher:     main(args)
2024-08-23T17:33:47.368620Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/tgi_service.py", line 16, in main
2024-08-23T17:33:47.368621Z DEBUG text_generation_launcher:     server.serve(
2024-08-23T17:33:47.368623Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/server.py", line 256, in serve
2024-08-23T17:33:47.368625Z DEBUG text_generation_launcher:     asyncio.run(
2024-08-23T17:33:47.368627Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
2024-08-23T17:33:47.368629Z DEBUG text_generation_launcher:     return loop.run_until_complete(main)
2024-08-23T17:33:47.368631Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
2024-08-23T17:33:47.368633Z DEBUG text_generation_launcher:     self.run_forever()
2024-08-23T17:33:47.368634Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
2024-08-23T17:33:47.368636Z DEBUG text_generation_launcher:     self._run_once()
2024-08-23T17:33:47.368640Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/base_events.py", line 1909, in _run_once
2024-08-23T17:33:47.368642Z DEBUG text_generation_launcher:     handle._run()
2024-08-23T17:33:47.368644Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
2024-08-23T17:33:47.368646Z DEBUG text_generation_launcher:     self._context.run(self._callback, *self._args)
2024-08-23T17:33:47.368648Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/grpc_interceptor/server.py", line 165, in invoke_intercept_method
2024-08-23T17:33:47.368652Z DEBUG text_generation_launcher:     return await self.intercept(
2024-08-23T17:33:47.368655Z DEBUG text_generation_launcher: > File "/usr/local/lib/python3.10/dist-packages/text_generation_server/interceptor.py", line 25, in intercept
2024-08-23T17:33:47.368657Z DEBUG text_generation_launcher:     return await response
2024-08-23T17:33:47.368660Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/opentelemetry/instrumentation/grpc/_aio_server.py", line 82, in _unary_interceptor
2024-08-23T17:33:47.368663Z DEBUG text_generation_launcher:     raise error
2024-08-23T17:33:47.368665Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/opentelemetry/instrumentation/grpc/_aio_server.py", line 73, in _unary_interceptor
2024-08-23T17:33:47.368668Z DEBUG text_generation_launcher:     return await behavior(request_or_iterator, context)
2024-08-23T17:33:47.368670Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/server.py", line 102, in Warmup
2024-08-23T17:33:47.368673Z DEBUG text_generation_launcher:     max_supported_total_tokens = self.model.warmup(request)
2024-08-23T17:33:47.368674Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 1108, in warmup
2024-08-23T17:33:47.368676Z DEBUG text_generation_launcher:     raise RuntimeError(
2024-08-23T17:33:47.368679Z DEBUG text_generation_launcher: RuntimeError: Not enough memory to handle 1 prefill tokens. You need to decrease `--max-batch-prefill-tokens`
2024-08-23T17:33:47.368683Z DEBUG text_generation_launcher: Method Warmup encountered an error.
2024-08-23T17:33:47.368686Z DEBUG text_generation_launcher: Traceback (most recent call last):
2024-08-23T17:33:47.368688Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 1106, in warmup
2024-08-23T17:33:47.368691Z DEBUG text_generation_launcher:     _, prefill_batch, _ = self.generate_token([batches[0]], is_warmup)
2024-08-23T17:33:47.368693Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/contextlib.py", line 79, in inner
2024-08-23T17:33:47.368696Z DEBUG text_generation_launcher:     return func(*args, **kwds)
2024-08-23T17:33:47.368698Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 897, in generate_token
2024-08-23T17:33:47.368701Z DEBUG text_generation_launcher:     batch.logits, batch.past = self.forward(
2024-08-23T17:33:47.368703Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 783, in forward
2024-08-23T17:33:47.368706Z DEBUG text_generation_launcher:     model_inputs = self.model.prepare_inputs_for_generation(**kwargs)
2024-08-23T17:33:47.368708Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/custom_modeling/llava_next.py", line 171, in prepare_inputs_for_generation
2024-08-23T17:33:47.368711Z DEBUG text_generation_launcher:     image_features = self.vision_tower(
2024-08-23T17:33:47.368714Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368716Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368718Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368722Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368724Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 339, in forward
2024-08-23T17:33:47.368727Z DEBUG text_generation_launcher:     return self.vision_model(
2024-08-23T17:33:47.368729Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368732Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368734Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368813Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368816Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 300, in forward
2024-08-23T17:33:47.368819Z DEBUG text_generation_launcher:     encoder_outputs = self.encoder(
2024-08-23T17:33:47.368821Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368823Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368826Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368828Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368831Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 251, in forward
2024-08-23T17:33:47.368833Z DEBUG text_generation_launcher:     layer_outputs = encoder_layer(
2024-08-23T17:33:47.368836Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368838Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368841Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368843Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368846Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 190, in forward
2024-08-23T17:33:47.368849Z DEBUG text_generation_launcher:     hidden_states, attn_weights = self.self_attn(
2024-08-23T17:33:47.368851Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1535, in _wrapped_call_impl
2024-08-23T17:33:47.368854Z DEBUG text_generation_launcher:     return self._call_impl(*args, **kwargs)
2024-08-23T17:33:47.368856Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1585, in _call_impl
2024-08-23T17:33:47.368859Z DEBUG text_generation_launcher:     result = forward_call(*args, **kwargs)
2024-08-23T17:33:47.368862Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/optimum/habana/transformers/models/clip/modeling_clip.py", line 158, in forward
2024-08-23T17:33:47.368864Z DEBUG text_generation_launcher:     attn_output = attn_output.reshape(bsz, tgt_len, embed_dim)
2024-08-23T17:33:47.368866Z DEBUG text_generation_launcher: RuntimeError: shape '[3, 577, 1024]' is invalid for input of size 886272
2024-08-23T17:33:47.368869Z DEBUG text_generation_launcher: 
2024-08-23T17:33:47.368871Z DEBUG text_generation_launcher: During handling of the above exception, another exception occurred:
2024-08-23T17:33:47.368874Z DEBUG text_generation_launcher: 
2024-08-23T17:33:47.368877Z DEBUG text_generation_launcher: Traceback (most recent call last):
2024-08-23T17:33:47.368879Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/tgi_service.py", line 37, in <module>
2024-08-23T17:33:47.368882Z DEBUG text_generation_launcher:     main(args)
2024-08-23T17:33:47.368884Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/tgi_service.py", line 16, in main
2024-08-23T17:33:47.368887Z DEBUG text_generation_launcher:     server.serve(
2024-08-23T17:33:47.368889Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/server.py", line 256, in serve
2024-08-23T17:33:47.368891Z DEBUG text_generation_launcher:     asyncio.run(
2024-08-23T17:33:47.368893Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
2024-08-23T17:33:47.368896Z DEBUG text_generation_launcher:     return loop.run_until_complete(main)
2024-08-23T17:33:47.368899Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
2024-08-23T17:33:47.368901Z DEBUG text_generation_launcher:     self.run_forever()
2024-08-23T17:33:47.368904Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
2024-08-23T17:33:47.368906Z DEBUG text_generation_launcher:     self._run_once()
2024-08-23T17:33:47.368907Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/base_events.py", line 1909, in _run_once
2024-08-23T17:33:47.368910Z DEBUG text_generation_launcher:     handle._run()
2024-08-23T17:33:47.368912Z DEBUG text_generation_launcher:   File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
2024-08-23T17:33:47.368915Z DEBUG text_generation_launcher:     self._context.run(self._callback, *self._args)
2024-08-23T17:33:47.368917Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/grpc_interceptor/server.py", line 165, in invoke_intercept_method
2024-08-23T17:33:47.368919Z DEBUG text_generation_launcher:     return await self.intercept(
2024-08-23T17:33:47.368922Z DEBUG text_generation_launcher: > File "/usr/local/lib/python3.10/dist-packages/text_generation_server/interceptor.py", line 25, in intercept
2024-08-23T17:33:47.368924Z DEBUG text_generation_launcher:     return await response
2024-08-23T17:33:47.368926Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/opentelemetry/instrumentation/grpc/_aio_server.py", line 82, in _unary_interceptor
2024-08-23T17:33:47.368928Z DEBUG text_generation_launcher:     raise error
2024-08-23T17:33:47.368930Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/opentelemetry/instrumentation/grpc/_aio_server.py", line 73, in _unary_interceptor
2024-08-23T17:33:47.368932Z DEBUG text_generation_launcher:     return await behavior(request_or_iterator, context)
2024-08-23T17:33:47.368935Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/server.py", line 102, in Warmup
2024-08-23T17:33:47.368937Z DEBUG text_generation_launcher:     max_supported_total_tokens = self.model.warmup(request)
2024-08-23T17:33:47.368940Z DEBUG text_generation_launcher:   File "/usr/local/lib/python3.10/dist-packages/text_generation_server/models/vlm_causal_lm.py", line 1108, in warmup
2024-08-23T17:33:47.368943Z DEBUG text_generation_launcher:     raise RuntimeError(
2024-08-23T17:33:47.368945Z DEBUG text_generation_launcher: RuntimeError: Not enough memory to handle 1 prefill tokens. You need to decrease `--max-batch-prefill-tokens`
Error: Warmup(Generation("CANCELLED"))

```

File: README.md [hunk 5]
```diff
@@ -284,6 +284,8 @@ curl -N 127.0.0.1:8080/generate_stream \
     -H 'Content-Type: application/json'
 ```
 
+Currently the model cannot support the multi-cards inference.
```
Comment (mandy-li) [inline README.md]: Currently Llava-v1.16 inference on multi-cards is not supported.

**Review by mandy-li — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -242,6 +250,68 @@ docker run -p 8080:80 \
...
+docker run -p 8080:80 \
+   --runtime=habana \
+   -v $volume:/data \
+   -v $PWD/quantization_config:/usr/src/quantization_config \
+   -v $PWD/hqt_output:/usr/src/hqt_output \
+   -e HABANA_VISIBLE_DEVICES=all \
+   -e OMPI_MCA_btl_vader_single_copy_mechanism=none \
+   -e HF_HUB_ENABLE_HF_TRANSFER=1 \
+   -e HUGGING_FACE_HUB_TOKEN=$hf_token \
+   -e QUANT_CONFIG=./quantization_config/maxabs_quant.json \
```
Comment (tthakkal) [inline README.md:322]:
```suggestion
   -e QUANT_CONFIG=./quantization_config/maxabs_quant.json \
   -e USE_FLASH_ATTENTION=true \
   -e FLASH_ATTENTION_RECOMPUTE=true \
```

File: README.md (hunk 5 above)
Comment (yuanwu2017) [reply README.md]: Done

**Review by mandy-li — APPROVED** [review]
LGTM. @regisss , please merge after PR#208

**Comment by yuanwu2017** [comment]
@regisss  Please help to review.

File: README.md [hunk 7]
```diff
@@ -265,7 +266,7 @@ docker run -p 8080:80 \
    --num-shard 8
 ```
 
-### LLava-next 7B BF16 on 1 Gaudi2 card
+### llava-v1.6-mistral-7b-hf BF16 on 1 Gaudi2 card
```
Comment (regisss) [inline README.md]: I think we should keep the former title as Llava 1.6 is Llava next and most people know it as Llava next.

File: README.md [hunk 8]
```diff
@@ -281,7 +282,44 @@ docker run -p 8080:80 \
...
+
+Send the simple request.
+```bash
+curl -N 127.0.0.1:8080/generate_stream \
+    -X POST \
+    -d '{"inputs":"![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/rabbit.png)What is this a picture of?\n\n","parameters":{"max_new_tokens":16, "seed": 42}}' \
+    -H 'Content-Type: application/json'
+```
+
+Currently Llava-v1.16 inference on multi-cards is not supported.
```
Comment (regisss) [inline README.md]:
```suggestion
Multi-card Llava-next inference is currently not supported.
```

File: README.md [hunk 9]
```diff
@@ -281,7 +282,44 @@ docker run -p 8080:80 \
...
+
+### llava-v1.6-mistral-7b-hf FP8 on 1 Gaudi2 card
```
Comment (regisss) [inline README.md]: Same

File: README.md (hunk 7 above)
Comment (yuanwu2017) [reply README.md]: Mandy said we may add other models of llava-next later, so we need to use specific names. For example, [llava-hf/llava-v1.6-vicuna-7b-hf](https://huggingface.co/llava-hf/llava-v1.6-vicuna-7b-hf)

Comment (regisss) [reply README.md]: In that case, maybe you can do:
```
### Llava-next

#### llava-v1.6-mistral-7b-hf BF16 on 1 Gaudi2 card

...

#### llava-v1.6-mistral-7b-hf FP8 on 1 Gaudi2 card
```

Comment (yuanwu2017) [reply README.md]: Done

**Review by regisss — APPROVED** [review]
LGTM!

**Pull request merged**